diff --git a/src/axi_demux_simple.sv b/src/axi_demux_simple.sv
index 1c11f79..62a538c 100644
--- a/src/axi_demux_simple.sv
+++ b/src/axi_demux_simple.sv
@@ -214,16 +214,16 @@ module axi_demux_simple #(
       ) i_aw_id_counter (
         .clk_i                        ( clk_i                          ),
         .rst_ni                       ( rst_ni                         ),
-        .lookup_axi_id_i              ( slv_req_i.aw.id[0+:AxiLookBits] ),
+        .lookup_axi_id_i              ( slv_req_i.aw.id& ((1 << AxiLookBits) - 1) ),
         .lookup_mst_select_o          ( lookup_aw_select               ),
         .lookup_mst_select_occupied_o ( aw_select_occupied             ),
         .full_o                       ( aw_id_cnt_full                 ),
         .inject_axi_id_i              ( '0                             ),
         .inject_i                     ( 1'b0                           ),
-        .push_axi_id_i                ( slv_req_i.aw.id[0+:AxiLookBits] ),
+        .push_axi_id_i                ( slv_req_i.aw.id& ((1 << AxiLookBits) - 1)),
         .push_mst_select_i            ( slv_aw_select_i                ),
         .push_i                       ( w_cnt_up                       ),
-        .pop_axi_id_i                 ( slv_resp_o.b.id[0+:AxiLookBits]  ),
+        .pop_axi_id_i                 ( slv_resp_o.b.id& ((1 << AxiLookBits) - 1) ),
         .pop_i                        ( slv_resp_o.b_valid & slv_req_i.b_ready      )
       );
       // pop from ID counter on outward transaction
@@ -359,16 +359,16 @@ module axi_demux_simple #(
       ) i_ar_id_counter (
         .clk_i                        ( clk_i                                       ),
         .rst_ni                       ( rst_ni                                      ),
-        .lookup_axi_id_i              ( slv_req_i.ar.id[0+:AxiLookBits]             ),
+        .lookup_axi_id_i              ( slv_req_i.ar.id & ((1 << AxiLookBits) - 1)         ),
         .lookup_mst_select_o          ( lookup_ar_select                            ),
         .lookup_mst_select_occupied_o ( ar_select_occupied                          ),
         .full_o                       ( ar_id_cnt_full                              ),
-        .inject_axi_id_i              ( slv_req_i.aw.id[0+:AxiLookBits]             ),
+        .inject_axi_id_i              ( slv_req_i.aw.id& ((1 << AxiLookBits) - 1)          ),
         .inject_i                     ( atop_inject                                 ),
-        .push_axi_id_i                ( slv_req_i.ar.id[0+:AxiLookBits]             ),
+        .push_axi_id_i                ( slv_req_i.ar.id & ((1 << AxiLookBits) - 1)        ),
         .push_mst_select_i            ( slv_ar_select_i                             ),
         .push_i                       ( ar_push                                     ),
-        .pop_axi_id_i                 ( slv_resp_o.r.id[0+:AxiLookBits]               ),
+        .pop_axi_id_i                 ( slv_resp_o.r.id& ((1 << AxiLookBits) - 1)         ),
         .pop_i                        ( slv_resp_o.r_valid & slv_req_i.r_ready & slv_resp_o.r.last )
       );
     end